pytest/conftest.py

```diff
@@ -122,6 +151,168 @@ def private_resource_with_metadata(sample_user):
     resource.delete()
 
 
+def create_composite_resource(u_name, u_email, u_lastname, u_firstname, res_title):
+    group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+    user = hydroshare.create_account(
+        u_email,
+        username=u_name,
+        first_name=u_firstname,
+        last_name=u_lastname,
+        superuser=False,
+        groups=[group]
+    )
+    _res = hydroshare.create_resource(
+        resource_type='CompositeResource',
+        owner=user,
+        title=res_title,
+        metadata=[],
+        files=()
+    )
+    return _res, user
+
+
+@pytest.mark.django_db
+@pytest.fixture(scope="function")
+def composite_resource():
+    """composite resource for testing"""
+    group, _ = Group.objects.get_or_create(name='Hydroshare Author')
+    user = hydroshare.create_account(
+        'user1@nowhere.com',
+        username='user1',
+        first_name='Creator_FirstName',
+        last_name='Creator_LastName',
+        superuser=False,
+        groups=[group]
+    )
+    _res = hydroshare.create_resource(
+        resource_type='CompositeResource',
+        owner=user,
+        title='Test Composite Resource',
+        metadata=[],
+        files=()
+    )
+
+    yield _res, user
+    _res.delete()
+    user.delete()
```

Comment:
What are the difference between create_composite_resource() and composite_resource()? It seems they are doing the same thing, and  create_composite_resource() can be replaced with composite_resource()

Reply:
There are differences. `create_composite_resource()` just a helper function to create a composite resource whereas `composite_resource()` is a test fixture which is executed anytime it is passed as a parameter to a test function. The `create_composite_resource()` helper function is used in another fixture `composite_resource_2()`.